src/AverageGlucoseService.js

```diff
@@ -1,26 +1,25 @@
-export const getTimeWeightedMean = (arrayOfGlucoseEvents) => {
-  const arrayOfRectangleSums = [];
+import { getStepWiseArray } from './StepWiseArrayService';
 
-  arrayOfGlucoseEvents.forEach((val, i) => {
-    if (i !== arrayOfGlucoseEvents.length - 1) {
-      const rectangle = getSegmentAverage(arrayOfGlucoseEvents[i], arrayOfGlucoseEvents[i+1]);
+export const getTimeWeightedMean = (glucoseEvents) => {
+  const sumOfGlucodeDurationProducts = getSumOfGlucodeDurationProducts(glucoseEvents);
+  const eventsDuration = getGlucoseEventsDuration(glucoseEvents);
 
-      arrayOfRectangleSums.push(rectangle);
-    }
-  });
-
-  const sumOfRectanges = arrayOfRectangleSums.reduce((cumulativeSum, rectangleSum) => (
-    cumulativeSum + rectangleSum
-  ), 0);
+  return sumOfGlucodeDurationProducts / eventsDuration;
+}
 
-  const timeDifference = arrayOfGlucoseEvents[arrayOfGlucoseEvents.length - 1].timestamp - arrayOfGlucoseEvents[0].timestamp;
+const getSumOfGlucodeDurationProducts = glucoseEvents => {
```

Comment:
Can we create a document or add some documentation to list and describe everything we are calculating?

Reply:
Yeah that's a great idea!